applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -67,6 +71,20 @@ void BaseSolidElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
             mConstitutiveLawVector.resize( integration_points.size() );
 
         InitializeMaterial();
+
+        if (mConstitutiveLawVector[0]->GetStrainSize() == 6) {
+            if (this->Has(LOCAL_AXIS_1) && this->Has(LOCAL_AXIS_2)) {
+                this->Set(BaseSolidElement::ROTATED, true);
+            } else {
+                this->Set(BaseSolidElement::ROTATED, false);
+            }
+        } else if (mConstitutiveLawVector[0]->GetStrainSize() == 3) {
+            if (this->Has(LOCAL_AXIS_1)) {
+                this->Set(BaseSolidElement::ROTATED, true);
+            } else {
+                this->Set(BaseSolidElement::ROTATED, false);
+            }
```

Comment:
I am not sure if this should be saved as a member.
- it is easy to compute
- adding members invalidates restart files / makes them unusable!